### PR TITLE
LPS-124271 Resolve computed image sized based on real media queries

### PR DESCRIPTION
**Pull request by p2kmgcl**

### CODE REVIEW

**Comment by p2kmgcl** [comment]
ci:test:sf

**Comment by p2kmgcl** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/page-structure/components/item-configuration-panels/ImageSourcePanel.js [hunk 1]
```diff
@@ -258,7 +258,9 @@ function DirectImagePanel({item}) {
 					fileEntryId={editableContent.fileEntryId}
 					imageSizeId={imageSizeId}
 					onImageSizeIdChanged={
-						type === EDITABLE_TYPES.image && handleImageSizeChanged
+						type === EDITABLE_TYPES.image
+							? handleImageSizeChanged
+							: null
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/page-structure/components/item-configuration-panels/ImageSourcePanel.js:263]: propTypes were failing?

Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/page-structure/components/item-configuration-panels/ImageSourcePanel.js:263]: More or less. If you pass false to an optional value it cries, but if you pass null it's fine :laughing:

**Comment by victorg1991** [comment]
LGTM, btw is this is a requirement of the story? Just wondering if we could just set the resolution based on the width from the viewports instead of relying on the current size of the window which would make things a bit simpler

**Comment by p2kmgcl** [comment]
> LGTM, btw is this is a requirement of the story? Just wondering if we could just set the resolution based on the width from the viewports instead of relying on the current size of the window which would make things a bit simpler

~Try if you dare~

Sadly it didn't because:
- Sometimes we do not have the "editableElement" (ex. container backgrounds)
- Media queries are checked against viewport size, not element width, so I found (well, Tim found) that the resolution is not the same than the real CSS.

But yes, this is terribly complex and I would prefer not to include it 
image

**Comment by victorg1991** [comment]
Ship it!

**Comment by victorg1991** [comment]
Maybe some poshi tests need adjustement though

**Comment by Tim-Cao** [comment]
Hi @p2kmgcl and @victorg1991, the Resolution field of Image fragment is not shown in non-desktop viewport when use direct Image Source. Could you have a look at?

**Comment by p2kmgcl** [comment]
On my way

**Comment by p2kmgcl** [comment]
@Tim-Cao I found to small issues, but the resolution is displayed correctly. Can you check with these changes?

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by Tim-Cao** [comment]
@p2kmgcl, works as expected

**Pull request closed without merging**